source/modulo_components/test/python/test_component.py

```diff
@@ -19,3 +22,19 @@ def test_add_output(component):
 
     component.add_output("test_13", "test", String)
     assert component._outputs["test_13"]["message_type"] == Bool
+
+
+def test_empty_parameter(component):
+    component.add_parameter(sr.Parameter("test", sr.ParameterType.DOUBLE_ARRAY), "description")
+    assert component.get_parameter("test").get_parameter_type() == sr.ParameterType.DOUBLE_ARRAY
+    assert Node.get_parameter(component, "test").type_ == Parameter.Type.NOT_SET
+
+    component.set_parameter_value("test", [1.0, 2.0], sr.ParameterType.DOUBLE_ARRAY)
+    assert Node.get_parameter(component, "test").type_ == Parameter.Type.DOUBLE_ARRAY
+
+    component.set_parameter_value("test", "test", sr.ParameterType.STRING)
+    assert component.get_parameter("test").get_parameter_type() == sr.ParameterType.DOUBLE_ARRAY
```

Comment:
maybe you can add some simple comments here to make it clear what the test is expecting. In this case the parameter validation fails and the type (and value) are not changed, but it's not particularly obvious